### PR TITLE
Implement RegExp flag accessors

### PR DESCRIPTION
**Pull request by HalidOdat**
It changes the following:
- Implement `RegExp.prototype.global` accessor
- Implement `RegExp.prototype.multiline` accessor
- Implement `RegExp.prototype.dotAll` accessor
- Implement `RegExp.prototype.ignoreCase` accessor
- Implement `RegExp.prototype.unicode` accessor
- Implement `RegExp.prototype.sticky` accessor
- Implement `RegExp.prototype.flags` accessor

### CODE REVIEW

**Review by RageKnify — APPROVED** [review]
Assume the panics are the 2 ghosts that keep showing up in CI.

File: boa/src/builtins/regexp/mod.rs [hunk 1]
```diff
@@ -207,76 +252,190 @@ impl RegExp {
...
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-get-regexp.prototype.unicode
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/unicode
+    pub(crate) fn get_unicode(this: &Value, _: &[Value], context: &mut Context) -> Result<Value> {
+        Self::regexp_has_flag(this, 'u', context)
+    }
+
+    /// `RegExp.prototype.sticky`
+    ///
+    /// The `flags` property returns a string consisting of the [`flags`][flags] of the current regular expression object.
```
Comment (Razican) [inline boa/src/builtins/regexp/mod.rs:371]: This seems to be missing a link to the `flags`. It seems it's failing to resolve the link.

File: boa/src/builtins/regexp/mod.rs [hunk 2]
```diff
@@ -289,39 +448,10 @@ impl RegExp {
     // ///
     // /// [spec]: https://tc39.es/ecma262/#sec-get-regexp.prototype.source
     // /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/source
-    // fn get_source(this: &Value, _: &[Value], _: &mut Context) -> Result<Value> {
+    // pub(crate) fn get_source(this: &Value, _: &[Value], _: &mut Context) -> Result<Value> {
```
Comment (Razican) [inline boa/src/builtins/regexp/mod.rs:451]: Why this change? Do we need this commented code?

**Review by Razican — COMMENTED** [review]
Looks good, just noticed a couple of things that might require a small fix.

File: boa/src/builtins/regexp/mod.rs (hunk 2 above)
Comment (HalidOdat) [reply boa/src/builtins/regexp/mod.rs:451]: I left it so we can uncomment it when we implement `get RegExp.prototype.source`

**Review by Razican — APPROVED** [review]
